src/Spectre.Cli/Internal/Extensions/TypeExtensions.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Linq;
+
+namespace Spectre.Cli
+{
+    internal static class TypeExtensions
+    {
+        public static bool IsFlagValue(this Type type)
+        {
+            return type.GetInterfaces().Any(i => i == typeof(IFlagValue));
+        }
+
+        public static Type GetFlagElementType(this Type type)
+        {
+            Type? current = type;
+            while (current != null)
+            {
+                if (current.IsGenericType && current.GetGenericTypeDefinition() == typeof(FlagValue<>))
+                {
+                    return current.GetGenericArguments().First();
+                }
+                current = current.BaseType;
+            }
+            throw new InvalidOperationException("Type is not a flag option.");
+        }
```

Comment:
Make this extension methods of `CommandOption` instead?

Reply:
+1 for not hanging the extension off `Type`